avocado/utils/process.py

```diff
@@ -348,6 +348,47 @@ def __init__(self, cmd, verbose=True, allow_output_check='all',
         self._popen = None
         self._ignore_bg_processes = ignore_bg_processes
 
+        def sigint_handler(signum, frame):
+            """
+            When we receive a SIGINT, we should just forward the same
+            SIGINT to the subprocess and, it that's enough to finish the
+            subprocess, we call the default_int_handler() to flag to our
+            caller that everything went as expected.
+            If subprocess keeps running after we send it a SIGINT, so do
+            we.
+            """
+            try:
+                self._popen.send_signal(signal.SIGINT)
+            except:
+                pass
+            if self._popen.poll() is not None:
+                self.result.interrupted = "signal/SIGINT"
+                signal.default_int_handler()
+
+        signal.signal(signal.SIGINT, sigint_handler)
+
+        def sigterm_handler(signum, frame):
+            """
+            When we receive a SIGTERM, it means we should really
+            terminate our subprocess. So, we first try to be nice and
+            send to the subprocess the same SIGTERM. If that's not
+            enough, we have to be more aggressive and kill it,
+            otherwise our caller will hang waiting our good will.
+            """
+            try:
+                self._popen.send_signal(signal.SIGTERM)
+            except:
+                pass
+
+            if self._popen.poll() is not None:
+                self.result.interrupted = "signal/SIGTERM"
+            else:
+                # No man left behind
+                self._popen.kill()
```

Comment:
I don't think you should automatically send kill when the process did not finish (yet). `process` library is a generic library and the user can send the `sigterm` manually, if required and when required (for example they expect the process to finish in 60s).

People can always use `proc.kill` to force-kill the process.